src/app/code/community/Shopgate/Cloudapi/Model/Api2/Carts/Url/Rest.php

```diff
@@ -37,15 +38,16 @@ protected function _create(array $filteredData)
      * @param Mage_Sales_Model_Quote $quote
      *
      * @return array
+     * @throws Mage_Core_Exception
      */
     protected function _createUrl($quote)
     {
         /** @var Shopgate_Cloudapi_Helper_Frontend_Checkout $helper */
         $helper = Mage::helper('shopgate_cloudapi/frontend_checkout');
-        $token  = $helper->generateAuthToken($quote->getId(), $quote->getCustomerId() ? : null);
+        $token  = $helper->generateAuthToken($quote->getId(), $this->_getStore(), $quote->getCustomerId());
```

Comment:
See comment above about return values of $this->_getStore()

Reply:
yes, I had a brain fart and thought I was in the controller operating with ```$request->getParam('store')``` call /facepalm